trade-finance-manager-ui/server/helpers/premium-payments-table-checkbox-id-helper.ts

```diff
@@ -0,0 +1,29 @@
+import { CURRENCY, Currency, FEE_RECORD_STATUS, FeeRecordStatus } from '@ukef/dtfs2-common';
+
+const CURRENCY_REGEX_GROUP = `(?<currency>${Object.values(CURRENCY).join('|')})`;
+const CURRENCY_REGEX = new RegExp(CURRENCY_REGEX_GROUP);
+
+const FEE_RECORD_STATUS_REGEX_GROUP = `(?<status>${Object.values(FEE_RECORD_STATUS).join('|')})`;
+const FEE_RECORD_STATUS_REGEX = new RegExp(FEE_RECORD_STATUS_REGEX_GROUP);
+
+const PREMIUM_PAYMENTS_TABLE_CHECKBOX_ID_REGEX = new RegExp(
+  `feeRecordId-\\d+-reportedPaymentsCurrency-${CURRENCY_REGEX_GROUP}-status-${FEE_RECORD_STATUS_REGEX_GROUP}`,
+);
+
+export const getPremiumPaymentsCheckboxIdsFromObjectKeys = (object: object): string[] =>
+  Object.keys(object).filter((key) => PREMIUM_PAYMENTS_TABLE_CHECKBOX_ID_REGEX.test(key));
+
+export const getFeeRecordStatusFromPremiumPaymentsCheckboxId = (checkboxId: string): FeeRecordStatus => {
```

Comment:
As we are using the non-null assertion on the result of `.match`, we should maybe think about narrowing the type of the input parameter to the `FeeRecordViewModelCheckboxId` type - otherwise, we may want to consider adding some error handling. What do you think?

Reply:
Yeah that makes sense, I have moved and renamed the type as now used fairly extensively outside the view model